### ISSUE TITLE
ProcessMonitor.active is actually Service.running

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1763]
|<img alt="soyt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| soyt reported|
|-|-|
|Trac ID|trac#1763|
|Type|enhancement|
|Created|2006-05-30 09:40:38Z|

It is a very minor cosmetic change, but ProcessMonitor.active is exactly what Service.running is there for, so it would make sense to use that. Patch follows.

Attachments:

* [procmon.py.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/da4/da4c8768d4def293944f4f613e3089bb4e42124e/524b341b077051cf0a9f7628ec10513bf5307623.diff) (2206 bytes) - added by soyt on 2006-05-30 09:41:37Z - 
* [test_procmon.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/da4/da4c8768d4def293944f4f613e3089bb4e42124e/10f25704953758188266a7dbf7713599f2aea903.py) (2827 bytes) - added by soyt on 2006-05-30 14:44:13Z - that is twisted.runner.test.test_procmon.py

<details><summary>Searchable metadata</summary>

```
trac-id__1763 1763
type__enhancement enhancement
reporter__soyt soyt
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__runner runner
keywords__ 
time__1148982038000000 1148982038000000
changetime__1273946028000000 1273946028000000
version__None None
owner__soyt soyt
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Note that twisted.runner has zero test coverage.  Adding some for the code that this patch changes would help get it applied faster.

**Comment by twisted-trac** [comment]
|<img alt="soyt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|soyt commented|
|-|-|

Ok, for what it is worth, I added a test case as a starting point.
Trial gives me nasty warnings about the reactor not ending in a proper state.
Somebody with better knowledge should really have a look.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|TimAllen commented|
|-|-|

These patches don't apply cleanly to today's trunk, and they conflict with changes made in [#3775](https://github.com/twisted/twisted/issues/3775).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

Removing review tag since there's some obvious work to be done (fix test, fix conflicts with trunk).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to soyt|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [28947]) Merge parameterize-procmon-3775-2 

Author: rwall
Reviewer: exarkun, jkakar, TimAllen
Fixes: [#3775](https://github.com/twisted/twisted/issues/3775)
Fixes: [#1763](https://github.com/twisted/twisted/issues/1763)

Add a `reactor` parameter to `twisted.runner.procmon.ProcessMonitor` to simplify
testing of code which uses that class.  Also remove a lot of unnecessary extra
code which is only necessary if process exited events get lost (which they ought
not to).
